go/test/endtoend/docker/vttestserver.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2021 The Vitess Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package docker
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+)
+
+const (
+	vttestserverMysql57image = "vttestserver-e2etest/mysql57"
+	vttestserverMysql80image = "vttestserver-e2etest/mysql80"
+)
+
+type vttestserver struct {
+	dockerImage          string
+	keyspaces            []string
+	numShards            []int
+	mysqlMaxConnecetions int
+	port                 int
+}
+
+func newVttestserver(dockerImage string, keyspaces []string, numShards []int, mysqlMaxConnections, port int) *vttestserver {
+	return &vttestserver{
+		dockerImage:          dockerImage,
+		keyspaces:            keyspaces,
+		numShards:            numShards,
+		mysqlMaxConnecetions: mysqlMaxConnections,
+		port:                 port,
+	}
+}
+
+func (v *vttestserver) teardown() {
+	cmd := exec.Command("docker", "rm", "--force", "vttestserver-end2end-test")
+	_ = cmd.Run()
```

Comment:
nit: should we not log something if the teardown fails?

Reply:
done